### PR TITLE
eslint-config: Lint all when .eslintignore changes

### PR DESCRIPTION
**Pull request by tbergq**

### CODE REVIEW

File: src/eslint-config-adeira/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Unreleased
 
 
+# 0.4.0
+
+- Lint all files when `.eslintignore` changes
```
Comment (mrtnzlml) [inline src/eslint-config-adeira/CHANGELOG.md]: We shouldn't write a changelog for versions before 1.0 - they can all be breaking by design and the scanner ignores most of the checks for such versions.

File: src/eslint-config-adeira/runner/__tests__/shouldLintAll.test.js [hunk 2]
```diff
@@ -13,6 +13,8 @@ test.each([
   'package.json',
   'src/packages/relay/package.json',
   '/Users/code/universe/src/packages/relay/package.json',
+  '.eslintignore',
+  '/Users/code/universe/src/packages/relay/.eslintignore',
```
Comment (mrtnzlml) [inline src/eslint-config-adeira/runner/__tests__/shouldLintAll.test.js]: Eslint ignore file which is not in the root doesn't work if I recall correctly.

File: src/eslint-config-adeira/runner/shouldLintAll.js [hunk 3]
```diff
@@ -3,5 +3,5 @@
 export default function shouldLintAll(filename: string): boolean %checks {
   // Eslint configs can be nested (not only the root path).
   // See: https://eslint.org/docs/user-guide/configuring#configuration-file-formats
-  return /\.eslintrc(?:\.(?:js(?:on)?|ya?ml))?$|package.json$/.test(filename);
+  return /\.eslintrc(?:\.(?:js(?:on)?|ya?ml))?$|package.json$|.eslintignore$/.test(filename);
```
Comment (mrtnzlml) [inline src/eslint-config-adeira/runner/shouldLintAll.js]: I have this feeling that we should fix this regexp to ignore dots. Something like this:

```diff
-   return /\.eslintrc(?:\.(?:js(?:on)?|ya?ml))?$|package.json$|.eslintignore$/.test(filename);
+   return /\.eslintrc(?:\.(?:js(?:on)?|ya?ml))?$|package\.json$|\.eslintignore$/.test(filename);
//                                                        ^       ^
```

File: src/eslint-config-adeira/runner/__tests__/shouldLintAll.test.js (hunk 2 above)
Comment (tbergq) [reply src/eslint-config-adeira/runner/__tests__/shouldLintAll.test.js]: Ahh, yes, you are right

File: src/eslint-config-adeira/runner/shouldLintAll.js (hunk 3 above)
Comment (tbergq) [reply src/eslint-config-adeira/runner/shouldLintAll.js]: You are right 🙈 Will do it right away

Comment (tbergq) [reply src/eslint-config-adeira/runner/shouldLintAll.js]: #178

**Pull request merged**